src/commands/flinkArtifacts.ts

```diff
@@ -1,9 +1,35 @@
-import { Disposable } from "vscode";
+import vscode, { Disposable } from "vscode";
 import { registerCommandWithLogging } from ".";
 import { ContextValues, setContextValue } from "../context/values";
 import { flinkArtifactUDFViewMode } from "../emitters";
+import { FlinkArtifact } from "../models/flinkArtifact";
 import { FlinkArtifactsViewProviderMode } from "../viewProviders/multiViewDelegates/constants";
 
+/**Open a new tab set to Flink SQL type with placeholder Flink UDF registration statement for selected artifact */
+export async function queryArtifactWithFlink(selectedArtifact: FlinkArtifact | undefined) {
+  if (!selectedArtifact || !(selectedArtifact instanceof FlinkArtifact)) {
```

Comment:
for now, because it shows up in the command palette. shouldn't ever happen though, so I stopped short of testing it